### PR TITLE
feat: support Image element prop for Avatar component

### PR DESCRIPTION
**Pull request by headwindz**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

[https://github.com/ant-design/ant-design/issues/27360](https://github.com/ant-design/ant-design/issues/27360)

### 💡 Background and solution

Add a new prop called `image`

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   support Image element as prop for Avatar component  |
| 🇨🇳 Chinese |  Avatar组件支持直接传入Image元素  |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/avatar/demo/type.md](https://github.com/n0ruSh/ant-design/blob/avatar/components/avatar/demo/type.md)
[View rendered components/avatar/index.en-US.md](https://github.com/n0ruSh/ant-design/blob/avatar/components/avatar/index.en-US.md)
[View rendered components/avatar/index.zh-CN.md](https://github.com/n0ruSh/ant-design/blob/avatar/components/avatar/index.zh-CN.md)

### CODE REVIEW

**Comment by xrkffgg** [comment]
feature branch

File: components/avatar/avatar.tsx [hunk 1]
```diff
@@ -29,6 +29,7 @@ export interface AvatarProps {
   className?: string;
   children?: React.ReactNode;
   alt?: string;
+  image?: React.ReactNode;
```
Comment (afc163) [inline components/avatar/avatar.tsx:32]: I prefer `<Avatar src={} />`

**Comment by afc163** [comment]
冲突了。

**Comment by afc163** [comment]
检查一下 ci，可以再 rebase 一下 feature。

**Comment by n0ruSh** [comment]
> 检查一下 ci，可以再 rebase 一下 feature。

@afc163 奇怪了。。重新rebase了一下feature分支。本地也跑了 npm run test -- -u 更新了snapshots. 但是 ci 里报了 Table 组件snapshot的测试用例失败。 这个可能是啥原因

**Comment by afc163** [comment]
重新安装 node_modules 后再跑 `npm run test -- -u`。

**Review by afc163 — APPROVED** [review]

**Pull request merged**